extensions/ql-vscode/src/data-extensions-editor/auto-model-usages-query.ts

```diff
@@ -0,0 +1,129 @@
+import { CancellationTokenSource } from "vscode";
+import { join } from "path";
+import { runQuery } from "./external-api-usage-query";
+import { CodeQLCliServer } from "../codeql-cli/cli";
+import { QueryRunner } from "../query-server";
+import { DatabaseItem } from "../databases/local-databases";
+import { interpretResultsSarif } from "../query-results";
+import { ProgressCallback } from "../common/vscode/progress";
+
+type Options = {
+  cliServer: Pick<
+    CodeQLCliServer,
+    "resolveDatabase" | "resolveQlpacks" | "interpretBqrsSarif"
+  >;
+  queryRunner: Pick<QueryRunner, "createQueryRun" | "logger">;
+  databaseItem: Pick<
+    DatabaseItem,
+    | "contents"
+    | "databaseUri"
+    | "language"
+    | "sourceArchive"
+    | "getSourceLocationPrefix"
+  >;
+  queryStorageDir: string;
+
+  progress: ProgressCallback;
+};
+
+export async function getAutoModelUsages({
+  cliServer,
+  queryRunner,
+  databaseItem,
+  queryStorageDir,
+  progress,
+}: Options): Promise<Record<string, string[]>> {
```

Comment:
Would it make sense to add type for signature rather than just using `string`?

Reply:
I don't think that will help since TypeScript doesn't strictly check type aliases (i.e. `type Signature = string`). If you pass a `string` where a `Signature` is required, it will allow it without any warning or error.

I have created a type for this return type so it's easier to know what it actually means.